packages/@react-native-windows/cli/src/generator-windows/index.ts

```diff
@@ -515,53 +516,46 @@ function toCppNamespace(namespace: string) {
   return namespace.replace(/\./g, '::');
 }
 
-export function installDependencies(options: {verbose: boolean}) {
-  const cwd = process.cwd();
-
-  // Extract react-native peer dependency version
-  const rnwPackageJsonPath = require.resolve(
-    'react-native-windows/package.json',
-    {paths: [process.cwd()]},
-  );
-  const rnwPackageJson = JSON.parse(
-    fs.readFileSync(rnwPackageJsonPath, {encoding: 'UTF8'}),
-  );
-  let rnPeerDependency = rnwPackageJson.peerDependencies['react-native'];
-  const depDelim = ' || ';
-  const delimIndex = rnPeerDependency.indexOf(depDelim);
-  if (delimIndex !== -1) {
-    rnPeerDependency = rnPeerDependency.slice(0, delimIndex);
+export async function installScriptsAndDependencies(options: {
+  verbose: boolean;
+}) {
+  const projectPackage = await WritableNpmPackage.fromPath(process.cwd());
+  if (!projectPackage) {
+    throw new Error('The current directory is not the root of an npm package');
```

Comment:
nit: Should we print what we think is the cwd? i.e.
```suggestion
    throw new Error(`The current directory '${process.cwd()}' is not the root of an npm package`);
```

Reply:
Can make a quick PR to do this in master